app/components/login/NewUserRegistration.js

```diff
@@ -0,0 +1,123 @@
+//Placeholder for newuser registration component
+
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { updateUser } from '../../redux/user';
+import { ErrorBar } from '../index';
+import Paper from '@material-ui/core/Paper';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import { useStyles } from '../index';
+
+function NewUserRegistration(props) {
+  const [loginError, setError] = useState(false);
+  const user = useSelector(state => state.user);
+  const [initLoad, setInit] = useState(true);
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const dispatch = useDispatch();
+
+  const classes = useStyles();
+
+  const handleChange = (event) => {
+    switch (event.target.id) {
+      case "firstname":
+        setFirstName(event.target.value);
+        break;
+      case "lastname":
+        setLastName(event.target.value);
+        break;
+        case "email":
+        setEmail(event.target.value);
+        break;
+        case "password":
+        setPassword(event.target.value);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = () => {
+    setInit(false);
+    setError(false);
+    dispatch(updateUser(user.id, { firstName, lastName, email, password, userId: user.id }));
```

Comment:
I recommend adding a check to see if the user email already exists in the database first

Reply:
The guest user is converted to actual user and email field will always be filled with dummy email per guest user creation flow, unless we keep it empty?? 
So check is not needed (?)

Reply:
Part of converting a guest to a real user should be asking the user to provide their real email. 
We should check if their real email already exists in the database, and return an error "you already have an account with this email"
If the email isn't there, then update the user record in the database with the real email.

Reply:
This only user creation screen and account management/edit should be separate functionality, not part of signup page ideally.

Reply:
if valid email exists check will have to be done. Will try if can be done tonight else will be done later.